### ISSUE TITLE
greedy_modularity_communities doesn't handle networks with string as node names

### ISSUE THREAD

**Issue by michelegentili93** [networkx/networkx#5006]
The nx.algorithms.community.modularity_max.greedy_modularity_communities function can't compute the clustering on networks with strings as nodes name. 

```
nt = nx.from_edgelist([['a','b'],['b','c'],['c','d'],['a','d'],['a','e']])
nx.algorithms.community.modularity_max.greedy_modularity_communities(nt,resolution=.5)
```
it would be enough to change `G.get_edge_data(i, j)` in modularity_max.py file using the name of the nodes rather than (I,j)

AttributeError                            Traceback (most recent call last)
<ipython-input-35-e6344b95bfb3> in <module>
      1 nt = nx.from_edgelist([['a','b'],['b','c'],['c','d'],['a','d'],['a','e']])
----> 2 nx.algorithms.community.modularity_max.greedy_modularity_communities(nt,resolution=.5)

~/.conda/envs/pypandaenv1/lib/python3.8/site-packages/networkx/algorithms/community/modularity_max.py in greedy_modularity_communities(G, weight, resolution)
     91     # H[n]: (-dq, i, j) for community with nth largest max_j(dQ_ij)
     92     a = [k[i] * q0 for i in range(N)]
---> 93     dq_dict = {
     94         i: {
     95             j: 2 * q0 * G.get_edge_data(i, j).get(weight, 1.0)

~/.conda/envs/pypandaenv1/lib/python3.8/site-packages/networkx/algorithms/community/modularity_max.py in <dictcomp>(.0)
     92     a = [k[i] * q0 for i in range(N)]
     93     dq_dict = {
---> 94         i: {
     95             j: 2 * q0 * G.get_edge_data(i, j).get(weight, 1.0)
     96             - 2 * resolution * k[i] * k[j] * q0 * q0

~/.conda/envs/pypandaenv1/lib/python3.8/site-packages/networkx/algorithms/community/modularity_max.py in <dictcomp>(.0)
     93     dq_dict = {
     94         i: {
---> 95             j: 2 * q0 * G.get_edge_data(i, j).get(weight, 1.0)
     96             - 2 * resolution * k[i] * k[j] * q0 * q0
     97             for j in [node_for_label[u] for u in G.neighbors(label_for_node[i])]

AttributeError: 'NoneType' object has no attribute 'get'

**Comment by rossbar** [comment]
Thanks for the report, this is related to #4966 which was fixed by #4965

**Comment by michelegentili93** [comment]
> Thanks for the report, this is related to #4966 which was fixed by #4965

I've upgraded nx to networkx-2.6.2 still not working.

**Comment by michelegentili93** [comment]
Okk! Thank you very much for the amazing work :)



Il giorno mer 4 ago 2021 alle ore 13:29 Ross Barnowski <
***@***.***> ha scritto:
